internal/plugin/plugin.go

```diff
@@ -126,21 +126,20 @@ func sortByDependencyOrder(original []*Plugin) ([]*Plugin, error) {
 
 // Load loads the buf.plugin.yaml at the specified path and returns a structure containing metadata for the plugin.
 func Load(path string, basedir string) (*Plugin, error) {
-	f, err := os.Open(path)
+	contents, err := os.ReadFile(path)
 	if err != nil {
 		return nil, err
 	}
-	defer f.Close()
 	var plugin Plugin
-	if err := yaml.NewDecoder(f).Decode(&plugin); err != nil {
-		return nil, err
-	}
 	plugin.Path = filepath.ToSlash(path)
 	plugin.Relpath, err = filepath.Rel(basedir, path)
 	if err != nil {
 		return nil, err
 	}
 	plugin.Relpath = filepath.ToSlash(plugin.Relpath)
+	if err := yaml.NewDecoder(bytes.NewReader(contents)).Decode(&plugin.ExternalConfig); err != nil {
```

Comment:
Do we want this to be a strict, or do we have checks earlier in the process that make sure the files are already valid? 

I.e., maybe we can use the config parser from bufbuild/buf?

Reply:
Switched to using encoding.UnmarshalJSONOrYAMLStrict. What I really want here is the external representation of the config, not the Plugin interface (which would require doing things like mapping to/from map/list options for tests).

Reply:
Me too.